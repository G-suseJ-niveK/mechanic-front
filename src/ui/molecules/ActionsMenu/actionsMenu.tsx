import React, { useCallback, ReactNode } from 'react';
import { MenuItem, IconButton, ListItemIcon, Icon, ListItemText } from '@material-ui/core';
import { Theme, makeStyles } from '@material-ui/core';
import MenuPopover from '~ui/atoms/MenuPopover/MenuPopover';
export type MenuListItem = {
  onClick(): void;
  icon: string;
  text: string;
};

type ActionsMenuProps = {
  listItems?: MenuListItem[];
  icon?: ReactNode;
};

// eslint-disable-next-line @typescript-eslint/no-unused-vars
const useStyles = makeStyles((theme: Theme) => ({
  menuItem: {
    padding: '15px',
    '&:hover': {
      backgroundColor: '#919EAB25'
    },
    '& .MuiListItemIcon-root': {
      minWidth: '35px'
    }
  }
}));

const ActionsMenu: React.FC<ActionsMenuProps> = (props: ActionsMenuProps) => {
  const classes = useStyles();
  const { listItems, icon } = props;
  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);

  const handleOnClick = useCallback((event: React.MouseEvent<HTMLButtonElement>) => {
    setAnchorEl(event.currentTarget);
  }, []);

  const handleClose = useCallback(() => {
    setAnchorEl(null);
  }, []);

  const handleMenuClick = useCallback((onClick: Function): void => {
    setAnchorEl(null);
    onClick();
  }, []);

  return (
    <>
      <IconButton size="small" onClick={handleOnClick}>
        {icon ? icon : <Icon style={{ color: '#446125' }}>more_horiz</Icon>}
      </IconButton>
      <MenuPopover sx={{ paddingTop: '7px' }} open={Boolean(anchorEl)} anchorEl={anchorEl} onClose={handleClose}>
        {listItems &&
          listItems.map((item: MenuListItem, index: number) => {
            return (
              <MenuItem
                key={`list_item_${index}`}
                onClick={() => handleMenuClick(item.onClick)}
                className={classes.menuItem}>
                {item.icon && (
                  <ListItemIcon>
                    <Icon>{item.icon}</Icon>
                  </ListItemIcon>
                )}
                <ListItemText primary={item.text} />
              </MenuItem>
            );
          })}
      </MenuPopover>
    </>
  );
};

export default React.memo(ActionsMenu);
