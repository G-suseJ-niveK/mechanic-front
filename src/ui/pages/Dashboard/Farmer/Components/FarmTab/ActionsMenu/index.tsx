import React, { useCallback } from 'react';
import { MenuItem, Button, ListItemText } from '@material-ui/core';
import { Theme, makeStyles } from '@material-ui/core';
import MenuPopover from '~ui/atoms/MenuPopover/MenuPopover';
import { Icon } from '@iconify/react';
import mapIcon from '@iconify/icons-ic/map';
import arrowIosDownwardOutline from '@iconify/icons-eva/arrow-ios-downward-outline';
import arrowIosUpwardFill from '@iconify/icons-eva/arrow-ios-upward-fill';

type ActionsMenuProps = {
  farms: any[];
  setFarm: any;
  value: string;
  currentTab: string;
  handleChangeTab: (value: string) => void;
};

// eslint-disable-next-line @typescript-eslint/no-unused-vars
const useStyles = makeStyles((theme: Theme) => ({
  menuItem: {
    padding: '15px',

    '&:hover': {
      backgroundColor: '#919EAB25'
    }
  }
}));

const ActionsMenu: React.FC<ActionsMenuProps> = (props: ActionsMenuProps) => {
  const classes = useStyles();
  const { farms, setFarm, value, handleChangeTab, currentTab } = props;
  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);

  const handleOnClick = useCallback((event: React.MouseEvent<HTMLButtonElement>) => {
    setAnchorEl(event.currentTarget);
  }, []);

  const handleClose = useCallback(() => {
    setAnchorEl(null);
  }, []);

  const handleMenuClick = useCallback((): void => {
    setAnchorEl(null);
  }, []);

  return (
    <>
      <Button
        onClick={handleOnClick}
        style={
          currentTab === value
            ? { color: '#212B36', marginRight: '40px', borderBottom: '2px solid #00AB55', borderRadius: '0px' }
            : { color: '#637381', marginRight: '40px', borderRadius: '0px' }
        }
        startIcon={<Icon icon={mapIcon} width={20} height={20} />}
        endIcon={
          <Icon icon={Boolean(anchorEl) ? arrowIosUpwardFill : arrowIosDownwardOutline} width={20} height={20} />
        }>
        Unidades productivas
      </Button>

      <MenuPopover sx={{ paddingTop: '7px' }} open={Boolean(anchorEl)} anchorEl={anchorEl} onClose={handleClose}>
        {farms?.map((item: any, index: number) => {
          return (
            <MenuItem
              key={`farms_item_${index}`}
              onClick={() => {
                setFarm(item);
                handleMenuClick();
                handleChangeTab(value);
              }}
              className={classes.menuItem}>
              <ListItemText primary={item?.name} />
            </MenuItem>
          );
        })}
      </MenuPopover>
    </>
  );
};

export default React.memo(ActionsMenu);
