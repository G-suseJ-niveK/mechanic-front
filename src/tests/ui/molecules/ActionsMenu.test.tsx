import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import ActionsMenu from '~molecules/ActionsMenu/actionsMenu';

describe('Render ActionsMenu', () => {
  test('should render', () => {
    const { container } = render(<ActionsMenu />);
    expect(container).toMatchSnapshot();
  });

  test('should render items', () => {
    const items = [
      {
        onClick: () => {
          //console.log('');
        },
        icon: 'mock-icon',
        text: 'mock-text'
      }
    ];
    render(<ActionsMenu listItems={items} />);
    fireEvent.click(screen.getByRole('button'));
    expect(screen.getByRole('menuitem', { name: 'mock-text' })).toBeInstanceOf(Node);
    expect(screen.getByText('mock-icon')).toBeInstanceOf(Node);
  });
});
