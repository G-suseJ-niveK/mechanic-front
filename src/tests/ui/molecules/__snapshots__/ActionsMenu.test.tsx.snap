// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render ActionsMenu should render 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
    tabindex="0"
    title="acciones"
    type="button"
  >
    <span
      class="MuiIconButton-label"
    >
      <span
        aria-hidden="true"
        class="material-icons MuiIcon-root"
        style="color: rgb(68, 97, 37);"
      >
        more_horiz
      </span>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
